feat: canonical categories

rbedeoan-plenty opened the pull request.

rbedeoan-plenty committed:
fix: canonical

rbedeoan-plenty committed:
feat: canoncial for category

rbedeoan-plenty committed:
fix: removed useCanonical

abocsan-plenty committed:
Merge branch 'main' into feat/canonical-categories

abocsan-plenty commented on apps/web/pages/category/[slug].vue:
```diff
@@ -37,8 +37,35 @@ const { data: categoryTree } = useCategoryTree();
 
+const setCanonical = () => {
+  const runtimeConfig = useRuntimeConfig();
+  const base = `${runtimeConfig.public.baseUrl}/c`;
```
You can use `runtimeConfig.public.apiUrl` directly.

abocsan-plenty commented on apps/web/pages/category/[slug].vue:
```diff
@@ -37,8 +37,35 @@ const { data: categoryTree } = useCategoryTree();
+};
 const handleQueryUpdate = async () => {
   await fetchProducts(getFacetsFromURL());
+  setCanonical();
```
Suggestion: to make this code cleaner, you can rename this method to `setCategoryCanonical` and move the body to the useCanonical composable.

abocsan-plenty reviewed (dismissed):
1. In the epic task ( https://dev.azure.com/plentymarkets/plentymarkets/_workitems/edit/67244 ) , at the Categories section, the link points to the category pagination metadata. This implementation does not account for that. Shouldn't it ?

2. The canonical url added to the page in LTS accounts for the filters in the url. Shouldn't this implementation also do that ? For example:     `<link rel="canonical" href="https://mevofvd5omld.c01-14.plentymarkets.com/living-room/?sorting=sorting.price.avg_asc">`

3. In the LTS I notice that there are no alternate links added when using pagination or url filters.

4. The LTS implementation has the `hreflang` attribute for the links. Shouldn't this implementation also account for that ?

rbedeoan-plenty committed:
fix: review

rbedeoan-plenty committed:
Merge branch 'main' into feat/canonical-categories

rbedeoan-plenty committed:
fix: hf

rbedeoan-plenty commented:
> 1. In the epic task ( https://dev.azure.com/plentymarkets/plentymarkets/_workitems/edit/67244 ) , at the Categories section, the link points to the category pagination metadata. This implementation does not account for that. Shouldn't it ?
Answer:
should be in the pagination task will add it to darius pr 
 <link rel="next" href="https://mevofvd5omld.c01-14.plentymarkets.com/living-room/?page=2" />

> 2. The canonical url added to the page in LTS accounts for the filters in the url. Shouldn't this implementation also do that ? For example:     `<link rel="canonical" href="https://mevofvd5omld.c01-14.plentymarkets.com/living-room/?sorting=sorting.price.avg_asc">`
yes, fixed

> 3. In the LTS I notice that there are no alternate links added when using pagination or url filters.
i don't think that will be a problem

> 4. The LTS implementation has the `hreflang` attribute for the links. Shouldn't this implementation also account for that ?
fixed

rbedeoan-plenty committed:
fix: review comments

rbedeoan-plenty committed:
fix: added prev and next

rbedeoan-plenty committed:
fix: lint

maxiroellplenty commented on apps/web/pages/category/[slug].vue:
```diff
@@ -38,8 +38,73 @@ const { data: categoryTree } = useCategoryTree();
 const { locale } = useI18n();
 const localePath = useLocalePath();
 
+const setCategoryCanonical = () => {
```
this function has to go into a composable.

rbedeoan-plenty replied:
i added it to useCanonical but it goes over the 15 line limit set in eslint? should i create a separate composable?

rbedeoan-plenty replied:
it is more comprehensive to have use canonical and all functions in there

maxiroellplenty replied:
I would split the composable method in two. So that you extract the category pagination into its own function.

rbedeoan-plenty committed:
Merge branch 'main' into feat/canonical-categories

rbedeoan-plenty committed:
fix: moved to composable

rbedeoan-plenty committed:
fix: added interfaces

rbedeoan-plenty committed:
fix: refactor

maxiroellplenty commented on apps/web/composables/useCanonical/types.ts:
```diff
@@ -5,10 +5,12 @@ export interface UseCanonicalState {
 }
 
 export type StaticPageMeta = () => void;
+export type CategoriesPageMeta = (productsCatalog: any, facetsFromUrl: any) => void;
```
use these types: import type { Facet } from '@plentymarkets/shop-api';
import { FacetSearchCriteria } from '@plentymarkets/shop-api'; instead of any :)

rbedeoan-plenty committed:
Merge branch 'main' into feat/canonical-categories

rbedeoan-plenty committed:
fix: replace any in types
